sdk-java/src/main/java/ly/count/sdk/java/internal/ModuleBackendMode.java

```diff
@@ -0,0 +1,434 @@
+    }
+
+    public Queue<Request> getRequestQ() {
+        return requestQ;
+    }
+
+    public Map<String, JSONArray> getEventQueues() {
+        return eventQueues;
+    }
+
+    public class BackendMode {
+        public void recordView(String deviceID, String key, Map<String, String> segmentation, long timestamp) {
+            if (!internalConfig.isBackendModeEnable()) {
+                L.wtf("[Countly][BackendMode][recordView] BackendMode is not enable.");
+                return;
+            }
+
+            if (deviceID == null || deviceID.isEmpty()) {
+                L.wtf("[Countly][BackendMode][recordView] DeviceID can not be null or empty.");
+                return;
+            }
+
+            if (key == null || key.isEmpty()) {
+                L.wtf("[Countly][BackendMode][recordView] Key can not be null or empty.");
+                return;
+            }
+
+            recordEventInternal(deviceID, key, 1, -1, -1, segmentation, timestamp);
+        }
+
+        public void recordEvent(String deviceID, String key, int count, double sum, double dur, Map<String, String> segmentation, long timestamp) {
+            if (!internalConfig.isBackendModeEnable()) {
+                L.wtf("[Countly][BackendMode][recordEvent] BackendMode is not enable.");
+                return;
+            }
+
+            if (deviceID == null || deviceID.isEmpty()) {
+                L.wtf("[Countly][BackendMode][recordEvent] DeviceID can not be null or empty.");
+                return;
+            }
+
+            if (key == null || key.isEmpty()) {
+                L.wtf("[Countly][BackendMode][recordEvent] Event key can not be null or empty.");
+                return;
+            }
+
+            recordEventInternal(deviceID, key, count, sum, dur, segmentation, timestamp);
+        }
+
+        public void sessionBegin(String deviceID, long timestamp) {
+            if (!internalConfig.isBackendModeEnable()) {
+                L.wtf("[Countly][BackendMode][recordEvent] BackendMode is not enable.");
+                return;
+            }
+
+            if (deviceID == null || deviceID.isEmpty()) {
+                L.wtf("[Countly][BackendMode][sessionBegin] DeviceID can not be null or empty.");
+                return;
+            }
+
+            Calendar calendar = Calendar.getInstance();
+            calendar.setTimeInMillis(timestamp);
+
+            final int hour = calendar.get(Calendar.HOUR_OF_DAY);
+            final int dow = calendar.get(Calendar.DAY_OF_WEEK) - 1;
+
+            Request request = new Request();
+            request.params.add("device_id", deviceID);
+            request.params.add("begin_session", 1);
+
+            request.params.add("dow", dow);
+            request.params.add("hour", hour);
+            request.params.add("timestamp", timestamp);
+            request.params.add("tz", DeviceCore.dev.getTimezoneOffset());
+
+//            JSONObject metrics = new JSONObject();
```

Comment:
Why do you leave this mess in?

Reply:
In case you ask me to add metrics in the session-begin call.

Reply:
sure, let's add metrics to the "session begin" call

Reply:
passed as a "string to string" map